src/core/public/core_app/core_app.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { UnregisterCallback } from 'history';
+import {
+  InternalApplicationSetup,
+  InternalApplicationStart,
+  AppNavLinkStatus,
+  AppMountParameters,
+} from '../application';
+import { HttpSetup, HttpStart } from '../http';
+import { CoreContext } from '../core_system';
+import { renderApp, setupUrlOverflowDetection } from './errors';
+import { NotificationsStart } from '../notifications';
+import { IUiSettingsClient } from '../ui_settings';
+
+interface SetupDeps {
+  application: InternalApplicationSetup;
+  http: HttpSetup;
+}
+
+interface StartDeps {
+  application: InternalApplicationStart;
+  http: HttpStart;
+  notifications: NotificationsStart;
+  uiSettings: IUiSettingsClient;
+}
+
+export class CoreApp {
+  private stopHistoryListening?: UnregisterCallback;
+
+  constructor(private readonly coreContext: CoreContext) {}
+
+  public setup({ http, application }: SetupDeps) {
+    application.register(this.coreContext.coreId, {
+      id: 'error',
+      title: 'App Error',
+      navLinkStatus: AppNavLinkStatus.hidden,
+      mount(params: AppMountParameters) {
+        return renderApp(params, { basePath: http.basePath });
```

Comment:
I thought about it, but since this is an error page that could be used for other things (for instance, when a app fails to mount), I don't want an networking problem from preventing this page displaying. Also it's extremely tiny.

I'll add a comment explaining that rationale.